src/iterable-functions.php

```diff
@@ -28,6 +28,21 @@ function iterable_map(iterable $iterable, callable $mapper): iterable
     return iterable($iterable)->map($mapper);
 }
 
+/**
+ * Merge one or more iterables
+ *
+ * @param iterable<TKey, TValue> $iterable
+ * @param iterable<iterable<TKey, TValue>> ...$args
+ * @return iterable<TKey, TValue>
+ *
+ * @template TKey
+ * @template TValue
+ */
+function iterable_merge(iterable $iterable, iterable ...$args): iterable
```

Comment:
As of PHP 7.4, `array_merge()` can be used without arguments (it basically returns an empty array). I'd suggest doing the same here.